File: scratch/compare_i18n.py
import json
import os

def get_keys(data, prefix=''):
    keys = set()
    if isinstance(data, dict):
        for k, v in data.items():
            new_prefix = f"{prefix}.{k}" if prefix else k
            keys.add(new_prefix)
            keys.update(get_keys(v, new_prefix))
    elif isinstance(data, list):
        # We don't usually care about list indices for simple key check, 
        # but next-intl raw might use them. For now, let's just track the list itself.
        pass
    return keys

path = 'c:/Users/Intel/OneDrive/Desktop/Projetos Github/victor-martins-portfolio/messages'
pt = json.load(open(os.path.join(path, 'pt.json'), encoding='utf-8'))
en = json.load(open(os.path.join(path, 'en.json'), encoding='utf-8'))
es = json.load(open(os.path.join(path, 'es.json'), encoding='utf-8'))

pt_keys = get_keys(pt)
en_keys = get_keys(en)
es_keys = get_keys(es)

print("Missing in EN:")
for k in sorted(pt_keys - en_keys):
    print(f"  {k}")

print("\nMissing in ES:")
for k in sorted(pt_keys - es_keys):
    print(f"  {k}")

print("\nExtra in EN (might be old keys):")
for k in sorted(en_keys - pt_keys):
    print(f"  {k}")

print("\nExtra in ES (might be old keys):")
for k in sorted(es_keys - pt_keys):
    print(f"  {k}")

File: scratch/compare_i18n.js
const fs = require('fs');

function getKeys(obj, prefix = '') {
    return Object.keys(obj).reduce((res, el) => {
        if (Array.isArray(obj[el])) {
            res.push(prefix + el);
        } else if (typeof obj[el] === 'object' && obj[el] !== null) {
            res.push(...getKeys(obj[el], prefix + el + '.'));
        } else {
            res.push(prefix + el);
        }
        return res;
    }, []);
}

const pt = JSON.parse(fs.readFileSync('messages/pt.json', 'utf8'));
const en = JSON.parse(fs.readFileSync('messages/en.json', 'utf8'));
const es = JSON.parse(fs.readFileSync('messages/es.json', 'utf8'));

const ptKeys = getKeys(pt);
const enKeys = getKeys(en);
const esKeys = getKeys(es);

console.log('--- Missing in EN ---');
ptKeys.filter(k => !enKeys.includes(k)).forEach(k => console.log(k));

console.log('\n--- Missing in ES ---');
ptKeys.filter(k => !esKeys.includes(k)).forEach(k => console.log(k));

console.log('\n--- Extra in EN ---');
enKeys.filter(k => !ptKeys.includes(k)).forEach(k => console.log(k));

console.log('\n--- Extra in ES ---');
esKeys.filter(k => !ptKeys.includes(k)).forEach(k => console.log(k));
